include/aligator/core/traj-opt-problem.hxx

```diff
@@ -173,4 +171,12 @@ void TrajOptProblemTpl<Scalar>::replaceStageCircular(
   stages_.pop_back();
 }
 
+template <typename Scalar>
+bool TrajOptProblemTpl<Scalar>::setInitCondIsStateError() const {
+  auto const* p = &*init_constraint_;
+  if (p == nullptr)
+    return false;
```

Comment:
If `p` is indeed nullptr, I think line 176 is actually undefined behaviour because it would deref the (empty) control block of the `polymorphic` type.
I'd suggest the following:
```suggestion
  if (init_constraint_.valueless_after_move())
    return false;
```